components/brave_rewards/core/contribution/contribution.cc

```diff
@@ -530,78 +527,72 @@ void Contribution::CreateNewEntry(const std::string& wallet_type,
 
   contribution->publishers = std::move(publisher_list);
 
-  auto shared_queue =
-      std::make_shared<mojom::ContributionQueuePtr>(std::move(queue));
-
-  auto save_callback = std::bind(&Contribution::OnEntrySaved, this, _1,
-                                 contribution->contribution_id, wallet_type,
-                                 *balance, shared_queue);
-
-  engine_->database()->SaveContributionInfo(contribution->Clone(),
-                                            save_callback);
+  engine_->database()->SaveContributionInfo(
+      contribution->Clone(),
```

Comment:
Could be moved if you saved `contribution->contribution_id` to a local variable.

Reply:
I wanted to err on the side of caution with respect to use after move, and I became exhausted at a certain point in trying to reason through and correct every call.